### DISCUSSION TITLE
How to extract a "status" report?

### DISCUSSION THREAD

**Discussion by KnBrckr** [nvim-orgmode/orgmode#715] [Q&A] [answered]
I'd like to generate a report (to use as input into a status report to team) of Active TODOs along with those completed in the last week. I've seen lisp code to accomplish something like this with emacs (for example https://stackoverflow.com/questions/22394394/orgmode-a-report-of-tasks-that-are-done-within-the-week) but haven't been able to work out how to do something like this using this plugin.

One aspect of the goal is to collect copies of TODOs, along with the related sub-content into a new buffer.  The only way I see to do this right now is a manual process. 

Advanced search looks like it has potential to find recently completed tasks where the help mentions using a search like "AGE<10", but this appears to require manually adding and maintaining such a property on all task entries.

But even with the search, it only shows the headline. To get the related content, it's a process of going to each item, and copying the headline + content to a "report" buffer.

Any pointers or thoughts on how to accomplish my goal in a more automated way? I've been poking at the API but haven't been able to wrap my head around it yet.

**Comment by kristijanhusak** [comment, marked as answer]
You can use [orgmode API](https://github.com/nvim-orgmode/orgmode/blob/master/doc/orgmode_api.txt) for that.

This can be your starting point:

```lua
local orgmode = require('orgmode.api')

local files = orgmode.load()

for _, file in ipairs(files) do
  for _, headline in ipairs(file.headlines) do
    print('headline', vim.inspect(headline)
  end
end
```

body of a headline is not provided, but you have the first line, and some details line deadline, schedule, etc.
If you think you could use something else I can extend api.

**Reply by KnBrckr** [reply]
Events took me in other directions for a while and now getting back to this. Thank you for this pointer. I've been able to get the core data collection working. With a slight change in behavior to use headlines to record anything that belongs in the resulting report, the ability to collect sub-headlines is a reasonable workaround.
